### PR TITLE
Use generated lists for DNR rules

### PR DESCRIPTION
**Pull request by smalluban**
Fixes https://github.com/ghostery/ghostery-dnr-lists/issues/17

This is not complete, as I still keep getting errors like "Failed to apply rules...", but as I get false positives, I can't get what rules are breaking it.

### CODE REVIEW

**Comment by chrmod** [comment]
@smalluban we probably should bump a minimum version of MacOS/iOS as urlFilters were added fairly recently AFAICT

**Comment by smalluban** [comment]
According to [MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/declarativeNetRequest#browser_compatibility) we can set the minimum version to 15.4 for iOS.

However, for the macOS, it is not that trivial. The Safari version is not tight with the OS version, so it is very likely that someone who has macOS 12 can have the latest version of Safari. Moreover, there are laptops that ended support for newer versions of macOS, but they still might receive Safari updates (this is my guess).

File: extension-manifest-v3/scripts/download-engines/utils.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    rule.condition.regexFilter?.match(/(\{\d*,\d*\}|\{\d\}|\|)/) ||
+    !supportedActions.includes(rule.action.type) ||
+    (resourceTypes && resourceTypes.length === 0)
+  ) {
+    return null;
+  }
+
+  return {
+    priority: rule.priority,
+    ...(rule.action.type === 'allowAllRequests'
```
Comment (chrmod) [inline extension-manifest-v3/scripts/download-engines/utils.js:46]: should we add some unit tests for all that?

**Review by chrmod — APPROVED** [review]

File: extension-manifest-v3/scripts/download-engines/utils.js (hunk 1 above)
Comment (smalluban) [reply extension-manifest-v3/scripts/download-engines/utils.js:46]: Finally, this should be taken out from the repository to https://github.com/ghostery/adblocker-filters. We can keep it here to test out with upcoming release and then move it out.

**Pull request merged**